### PR TITLE
fix(onStartTyping): Incorrect accepted valid characters

### PR DESCRIPTION
**Pull request by Bernard-Borg**
### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

---

### Description

This PR fixes the characters which are accepted when using onStartTyping. Specifically, it adds back the ability for users to start typing with Numpad0 to focus on the connected element (#4614, related: #2932).

### Additional context

Should more characters be accepted or are just the basic letters and numbers enough?

### CODE REVIEW

File: packages/core/onStartTyping/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  let element: Ref<HTMLInputElement>
+  let callBackFn: any
+
+  beforeEach(() => {
+    element = shallowRef(document.createElement('input'))
+    element.value.tabIndex = 1
+    callBackFn = vi.fn()
+  })
+
+  function createKeyDownEvent(keyCode: number) {
```
Comment (43081j) [inline packages/core/onStartTyping/index.test.ts:16]: just a tiny nit, can you call this `dispatchKeyDownEvent` to make it clear it not only creates it, but also dispatches it

**Review by 43081j — CHANGES_REQUESTED** [review]
can you add one more test 👀  a negative

e.g. if you type arrow keys or some such thing, we _do not_ trigger the callback

File: packages/core/onStartTyping/index.test.ts (hunk 1 above)
Comment (Bernard-Borg) [reply packages/core/onStartTyping/index.test.ts]: Done

**Review by 43081j — APPROVED** [review]

**Review by OrbisK — APPROVED** [review]

**Review by ilyaliao — APPROVED** [review]
